storybookjs/storybook
# UI: Default white preview background

Closes #21080



## What I did

Set the preview iframe to default to a white background. Previously it was transparent, meaning that Storybook in dark mode would cause a dark background, which is unexpected given that browsers default to a white background - only the user's code should change the background color, not Storybook's code.

Setting the color on the iframe element ensures that user code that sets it on `html`, `body`, or something else **within** the iframe still overrides it.

I've tested that

1. this still works with the backgrounds addon
2. it's still possible for user code to change the background, even based on browser theme, with:

```css
@media screen and (prefers-color-scheme: dark) {
  body {
    background-color: darkblue;
    color: lightgray;
  }
}
```


## How to test

1. Open one of the published sandboxes (eg. this one:)
2. Navigate to one of the page stories:
3. Change your browser color scheme to dark (see screenshot) and reload
4. see that Storybook UI goes dark, but the preview stays white.

Feel free to test this out locally with custom css in stories.



## Checklist



      [MIGRATION.MD](https://github.com/storybookjs/storybook/blob/next/MIGRATION.md)

#### Maintainers

      make sure to add the `ci:merged` or `ci:daily` GH label to it.

`["cleanup", "BREAKING CHANGE", "feature request", "bug", "documentation", "maintenance", "dependencies", "other"]`
